includes/classes/RegisteredDataHandler.php

```diff
@@ -0,0 +1,637 @@
+	 * @param bool  $update_hierarchy Whether to update the term hierarchy.
+	 * @return int The term ID of the processed term.
+	 */
+	public function process_registered_data_term( $term_data, $process_parent = false, $update_hierarchy = false ) {
+		if ( empty( $term_data ) ) {
+			return 0;
+		}
+
+		if ( ! is_array( $term_data ) ) {
+			$term_data = (array) $term_data;
+		}
+
+		$process_parent = $process_parent && is_taxonomy_hierarchical( $term_data['taxonomy'] );
+		$parent_term_id = 0;
+		if ( $process_parent && ! empty( $term_data['parent'] ) ) {
+			if ( ! empty( $term_data['parent']['term_id'] ) ) {
+				$parent_term_id = $this->process_registered_data_term( $term_data['parent'], $process_parent, $update_hierarchy );
+			}
+		}
+
+		// Check if the term exists on the target site already and return the term ID if it does.
+		$taxonomy = $term_data['taxonomy'] ?? '';
+		$term     = get_term_by( 'slug', $term_data['slug'], $taxonomy );
+		if ( ! empty( $term ) && ! empty( $term->term_id ) ) {
+			if ( $update_hierarchy && $process_parent && ! empty( $parent_term_id ) && $term->parent !== $parent_term_id ) {
+				wp_update_term( $term->term_id, $taxonomy, array( 'parent' => $parent_term_id ) );
+			}
+
+			return $term->term_id;
+		}
+
+		$args = array(
+			'slug'        => $term_data['slug'],
+			'description' => $term_data['description'],
+		);
+
+		if ( $process_parent && ! empty( $parent_term_id ) ) {
+			$args['parent'] = $parent_term_id;
+		}
+
+		$term = wp_insert_term(
+			$term_data['name'],
+			$taxonomy,
+			$args
+		);
+
+		if ( is_wp_error( $term ) || empty( $term['term_id'] ) ) {
+			return 0;
+		}
+
+		return $term['term_id'];
+	}
+
+	/**
+	 * Pre-process registered "post" type data for external connections.
+	 *
+	 * This function handles "post" type data when pushing to an external connection.
+	 * Since the target site cannot pull the post, we push it to the target site,
+	 * retrieve the remote post ID, and include it in the extra data sent to the target site.
+	 *
+	 * @param array                           $post_data The post data to process for the external connection.
+	 * @param \Distributor\ExternalConnection $connection The connection object.
+	 * @return array The processed extra data.
+	 */
+	public function pre_process_registered_data_post( $post_data, $connection ) {
+		$extra_data = $post_data['distributor_extra_data'] ?? array();
+		if ( empty( $extra_data ) || ! is_array( $extra_data ) || ! apply_filters( 'dt_process_extra_data', true ) ) {
+			return $post_data;
+		}
+
+		try {
+			$registered_data      = distributor_get_registered_data();
+			$registered_post_data = array_filter(
+				$registered_data,
+				function( $arr ) {
+					return 'post' === $arr['type'];
+				}
+			);
+
+			if ( ! empty( $registered_post_data ) && ! empty( $extra_data ) ) {
+				// Disable the process_extra_data filter to prevent infinite loop.
+				add_filter( 'dt_process_extra_data', '__return_false' );
```

Comment:
This and the following remove_filter call present a risk if a developer tries to use the filter to disable processing by adding the `__return_false` callback. The code in this method will remove their filter. (Based on something [similar in WooCommerce](https://github.com/woocommerce/woocommerce/blob/8eeadc72263e5994b8f0bb1a3b5b0b69941dde83/plugins/woocommerce/src/Internal/DataStores/Orders/LegacyDataHandler.php#L325-L327))

```suggestion
				$prevent_processing = function(){ return false; };
				add_filter( 'dt_process_extra_data', $prevent_processing );
```

Reply:
Thanks for this note, handled in https://github.com/10up/distributor/pull/1309/commits/81ece4156f5c323815e9c836c7e5d62691ac6290